astropy/convolution/src/boundary_none.c

```diff
@@ -0,0 +1,513 @@
+    else
+        convolve2d_boundary_none(result, f, nx, ny, g, nkx, nky, false, n_threads);
+}
+
+void convolve3d_boundary_none_c(DTYPE * const result,
+        const DTYPE * const f, const size_t nx, const size_t ny, const size_t nz,
+        const DTYPE * const g, const size_t nkx, const size_t nky, const size_t nkz,
+        const bool nan_interpolate,
+        const unsigned n_threads)
+{
+#ifdef NDEBUG
+    if (!result || !f || !g)
+        return;
+#else
+    assert(result);
+    assert(f);
+    assert(g);
+#endif
+
+    if (nan_interpolate)
+        convolve3d_boundary_none(result, f, nx, ny, nz, g, nkx, nky, nkz, true, n_threads);
+    else
+        convolve3d_boundary_none(result, f, nx, ny, nz, g, nkx, nky, nkz, false, n_threads);
+}
+
+// 1D
+inline __attribute__((always_inline)) void convolve1d_boundary_none(DTYPE * const result,
+        const DTYPE * const f, const size_t _nx,
+        const DTYPE * const g, const size_t _nkx,
+        const bool _nan_interpolate,
+        const unsigned n_threads)
+{
+#ifdef NDEBUG
+    if (!result || !f || !g)
+        return;
+#else
+    assert(result);
+    assert(f);
+    assert(g);
+#endif
+
+    const unsigned _wkx = _nkx / 2;
+#ifdef NDEBUG
+    if (!(_nx > 2*_wkx))
+        return;
+#else
+    assert(_nx > 2*_wkx);
+#endif
+
+#ifdef _OPENMP
+    omp_set_num_threads(n_threads); // Set number of threads to use
+#pragma omp parallel
+    { // Code within this block is threaded
+#endif
+
+    // Copy these to thread locals to allow compiler to optimize (hoist/loads licm)
+    // when threaded. Without these, compile time constant conditionals may
+    // not be optimized away.
+    const size_t nx = _nx;
+    const size_t nkx = _nkx;
+    const bool nan_interpolate = _nan_interpolate;
+
+    // Thread locals
+    const unsigned wkx = _wkx;
+    const unsigned nkx_minus_1 = nkx-1;
+    unsigned wkx_minus_i;
+    unsigned ker_i;
+    const unsigned nx_minus_wkx = nx - wkx;
+    unsigned i_minus_wkx;
+    const unsigned wkx_plus_1 = wkx + 1;
+    unsigned i_plus_wkx_plus_1;
+    unsigned nkx_minus_1_minus_wkx_plus_i;
+
+    DTYPE top, bot=0., ker, val;
+
+#ifdef _OPENMP
+#pragma omp for schedule(dynamic)
+#endif
+    for (unsigned i = wkx; i < nx_minus_wkx; ++i)
+    {
+        wkx_minus_i = wkx - i; // wkx - 1
+        i_minus_wkx = i - wkx; //i - wkx
+        i_plus_wkx_plus_1 = i + wkx_plus_1; // i + wkx + 1
+        nkx_minus_1_minus_wkx_plus_i = nkx_minus_1 - wkx_minus_i; // nkx - 1 - (wkx - i)
+
+        top = 0.;
+        if (nan_interpolate) // compile time constant
+            bot = 0.;
+        for (unsigned ii = i_minus_wkx; ii < i_plus_wkx_plus_1; ++ii)
+        {
+            ker_i = nkx_minus_1_minus_wkx_plus_i - ii; // nkx - 1 - (wkx + ii - i)
+            val = f[ii];
+            ker = g[ker_i];
+            if (nan_interpolate) // compile time constant
+            {
+                if (!isnan(val))
```

Comment:
Is there a performance cost to this `if` condition within (what I assume) is a hot-loop? For instance, if the code is re-written as:

```C
    DTYPE nanval = isnan(val) && nan_interpolate ? 0.0:1.0;
    top += nanval * val * ker;
    bot += nanval * ker;
```

(please check that logic is correct)

Reply:
Oops. Apparently I mid-advertently clicked the review button :(